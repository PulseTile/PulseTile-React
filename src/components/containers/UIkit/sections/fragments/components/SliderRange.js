import React, { Component} from 'react';
import { Range } from 'rc-slider';

export default class SliderRange extends Component {

  state = {
    rangeForm: null,
    rangeTo: null,
  };

  changeRange = (value) => {
    this.setState({
      rangeForm: value[0],
      rangeTo: value[1]
    });
  };

  getMarksArray = (minValueRange, maxValueRange) => {
    const marks = {};
    const rangeAmountPart = 20;
    const different = maxValueRange - minValueRange;
    const markWeight = parseInt(different / rangeAmountPart);
    marks[minValueRange] = minValueRange;
    marks[maxValueRange] = maxValueRange;
    for (let i = 1; i < rangeAmountPart; i++) {
      const mark = minValueRange + markWeight * i;
      const markLabel = (0 === i % 2) ? mark : '';
      marks[mark] = { label: markLabel };
    }
    return marks;
  };

  render () {
    const minValueRange = 0;
    const maxValueRange = 100;
    return (
      <div id="slider-range" className="ui-section">
        <strong className="ui-title">Slider Range</strong>
        <div className="form-group-wrapper">
          <div className="form-group">
            <label className="control-label">Range (Years)</label>
            <div className="wrap-rzslider">
              <div className="wrap-rzslider-events">
                <Range
                  min={minValueRange}
                  max={maxValueRange}
                  defaultValue={[minValueRange, maxValueRange]}
                  marks={this.getMarksArray(minValueRange, maxValueRange)}
                  tipFormatter={value => value}
                  tipProps={{ placement: 'top', visible: true, defaultVisible: true }}
                  onChange={this.changeRange}
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    );
  }
};
