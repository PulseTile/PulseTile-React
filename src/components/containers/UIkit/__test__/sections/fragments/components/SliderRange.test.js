import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';
import SliderRange from '../../../../sections/fragments/components/SliderRange';

Enzyme.configure({ adapter: new Adapter() });

describe('Component <SliderRange />', () => {
  it('should renders SliderRange with props correctly', () => {
    const component = shallow(<SliderRange />);
    expect(component).toMatchSnapshot();
  });
});
