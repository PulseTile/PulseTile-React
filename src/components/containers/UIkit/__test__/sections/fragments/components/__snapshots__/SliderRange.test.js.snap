// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component <SliderRange /> should renders SliderRange with props correctly 1`] = `
<div
  className="ui-section"
  id="slider-range"
>
  <strong
    className="ui-title"
  >
    Slider Range
  </strong>
  <div
    className="form-group-wrapper"
  >
    <div
      className="form-group"
    >
      <label
        className="control-label"
      >
        Range (Years)
      </label>
      <div
        className="wrap-rzslider"
      >
        <div
          className="wrap-rzslider-events"
        >
          <ComponentEnhancer(Range)
            activeDotStyle={Object {}}
            allowCross={true}
            className=""
            count={1}
            defaultValue={
              Array [
                0,
                100,
              ]
            }
            disabled={false}
            dotStyle={Object {}}
            dots={false}
            handle={[Function]}
            handleStyle={
              Array [
                Object {},
              ]
            }
            included={true}
            marks={
              Object {
                "0": 0,
                "10": Object {
                  "label": 10,
                },
                "100": 100,
                "15": Object {
                  "label": "",
                },
                "20": Object {
                  "label": 20,
                },
                "25": Object {
                  "label": "",
                },
                "30": Object {
                  "label": 30,
                },
                "35": Object {
                  "label": "",
                },
                "40": Object {
                  "label": 40,
                },
                "45": Object {
                  "label": "",
                },
                "5": Object {
                  "label": "",
                },
                "50": Object {
                  "label": 50,
                },
                "55": Object {
                  "label": "",
                },
                "60": Object {
                  "label": 60,
                },
                "65": Object {
                  "label": "",
                },
                "70": Object {
                  "label": 70,
                },
                "75": Object {
                  "label": "",
                },
                "80": Object {
                  "label": 80,
                },
                "85": Object {
                  "label": "",
                },
                "90": Object {
                  "label": 90,
                },
                "95": Object {
                  "label": "",
                },
              }
            }
            max={100}
            min={0}
            onAfterChange={[Function]}
            onBeforeChange={[Function]}
            onChange={[Function]}
            prefixCls="rc-slider"
            pushable={false}
            railStyle={Object {}}
            step={1}
            tipFormatter={[Function]}
            tipProps={
              Object {
                "defaultVisible": true,
                "placement": "top",
                "visible": true,
              }
            }
            trackStyle={
              Array [
                Object {},
              ]
            }
            vertical={false}
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
